src/main/java/com/ss/mqtt/broker/handler/packet/in/SubscribeInPacketHandler.java

```diff
@@ -15,5 +19,17 @@
     protected void handleImpl(@NotNull UnsafeMqttClient client, @NotNull SubscribeInPacket packet) {
         var ackReasonCodes = subscriptionService.subscribe(client, packet.getTopicFilters());
         client.send(client.getPacketOutFactory().newSubscribeAck(packet.getPacketId(), ackReasonCodes));
+        ackReasonCodes.findAny(client, SubscribeInPacketHandler::hasInvalidReason);
```

Comment:
you should take the 'ackCode' by condition and if it's not null then close this client inside this method, it's not correct to do it inside the search condition.